openeo/extra/job_management.py

```diff
@@ -458,3 +422,60 @@ def ignore_connection_errors(context: Optional[str] = None):
         _log.warning(f"Ignoring connection error (context {context or 'n/a'}): {e}")
         # Back off a bit
         time.sleep(5)
+
+
+class JobTrackerStorage:
+    # TODO: add support to store as Parquet file as well as CSV.
+    # TODO: add support to store to both PosixPath and URL.
+    """
+    Helper to manage the storage of batch job metadata.
+    """
+
+    def __init__(self):
+        pass
+
+    def normalize_df(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Ensure we have the required columns and the expected type for the geometry column.
+
+        :param df: The dataframe to normalize.
+        :return: a new dataframe that is normalized.
+        """
+
+        # check for some required columns.
+        required_with_default = [
+            ("status", "not_started"),
+            ("id", None),
+            ("start_time", None),
+            ("cpu", None),
+            ("memory", None),
+            ("duration", None),
+            ("backend_name", None),
+        ]
+        new_columns = {col: val for (col, val) in required_with_default if col not in df.columns}
+        df = df.assign(**new_columns)
+
+        return df
+    def resume_df(self, df: pd.DataFrame, output_file: Union[str, Path]) -> pd.DataFrame:
```

Comment:
I think the `output_file` should go in the `__init__`. Otherwise you can't work with alternative storage implementations that don't work with files